addons/website/static/src/js/editor/snippets.options.js

```diff
@@ -1767,6 +1767,39 @@ options.registry.CoverProperties = options.Class.extend({
     },
 });
 
+/**
+ * allows control over shadows
+ */
+options.registry.shadow = options.Class.extend({
+    /**
+     * @override
+     */
+    selectClass(previewMode, widgetValue, params) {
+        this._super(...arguments);
+        if (!widgetValue) {
+            this.$target.css('box-shadow', '');
+        }
+    },
+
+    //--------------------------------------------------------------------------
+    // Private
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     */
+    async _computeWidgetState(methodName, params) {
+        let value = await this._super(...arguments);
+        if (methodName === 'selectStyle' && params.cssProperty === 'box-shadow') {
+            const color = value.slice(value.indexOf('rgb'), value.indexOf(')') + 1);
+            const inset = value.indexOf('inset') >= 0 ? 'inset' : '';
+            value = value.replace(color, '').replace('inset', '');
+            value = `${color.replace(/\s/g, '')} ${value.trim()} ${inset}`;
+        }
+        return value;
+    },
```

Comment:
I'd integrate that in the base option class, so that `selectStyle` work with box-shadow in any option.